Source/projections/Builder/ProjectionClassBuilder.ts

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { Logger } from 'winston';
+
+import { EventType, EventTypeId, EventTypeMap, Generation, IEventTypes } from '@dolittle/sdk.artifacts';
+import { Cancellation } from '@dolittle/sdk.resilience';
+import { Guid } from '@dolittle/rudiments';
+import { IContainer } from '@dolittle/sdk.common';
+import { ExecutionContext } from '@dolittle/sdk.execution';
+import { Constructor } from '@dolittle/types';
+
+import { ProjectionsClient } from '@dolittle/runtime.contracts/Runtime/Events.Processing/Projections_grpc_pb';
+
+import { IProjections, ProjectionCallback, Projection, ProjectionResult } from '../';
+
+
+import { CannotRegisterProjectionThatIsNotAClass } from './CannotRegisterProjectionThatIsNotAClass';
+import { CouldNotCreateInstanceOfProjection } from './CouldNotCreateInstanceOfProjection';
+import { ICanBuildAndRegisterAProjection } from './ICanBuildAndRegisterAProjection';
+import { OnDecoratedMethod } from './OnDecoratedMethod';
+import { OnDecoratedMethods } from './OnDecoratedMethods';
+import { on as onDecorator } from './onDecorator';
+import { ProjectionDecoratedType } from './ProjectionDecoratedType';
+import { ProjectionDecoratedTypes } from './ProjectionDecoratedTypes';
+import { projection as projectionDecorator } from './projectionDecorator';
+import { ReadModelAlreadyRegistered } from './ReadModelAlreadyRegistered';
+import { KeySelector } from '../KeySelector';
+import { ProjectionProcessor } from '../Internal/ProjectionProcessor';
+import { DeleteReadModelInstance } from '../DeleteReadModelInstance';
+
+export class ProjectionClassBuilder<T> implements ICanBuildAndRegisterAProjection {
+    private readonly _projectionType: Constructor<T>;
+    private readonly _getInstance: (container: IContainer) => T;
+
+    constructor(typeOrInstance: Constructor<T> | T) {
+        if (typeOrInstance instanceof Function) {
+            this._projectionType = typeOrInstance;
+            this._getInstance = container => container.get(typeOrInstance);
+
```

Comment:
Does `typeOrInstance instanceof Function` behave the same way as `typeof typeOrInstance === 'function'` ?

Reply:
For functions [yes](https://stackoverflow.com/a/6625960/5806412)